torch/nn/utils/rnn.py

```diff
@@ -124,3 +124,40 @@ def pad_packed_sequence(sequence, batch_first=False):
     if batch_first:
         output = output.transpose(0, 1)
     return output, lengths
+
+def get_last_step_indices(lengths):
+    """A helper function for :func:`get_last_step_tensor`.
+
+    The returned indices is used to select the last step of a :class:`PackedSequence` object.
+
+    Arguments:
+        lengths (list[int]): list of sequences lengths of each batch element.
+
+    Returns:
+        List of int containing the indices of the last step of each sequence.
+    """
+    indices = []
+    for i, length in enumerate(lengths):
+        index = sum([min(length - 1, l) for l in lengths]) + i
+        indices.append(index)
+    return indices
```

Comment:
This can be computed in O(n) time instead of O(n^2):
```python
n_lengths = len(lengths)
rev_lengths = lengths[::-1]
rev_lengths_sum = torch.LongTensor(rev_lengths).cumsum()
return torch.LongTensor([(n_lengths - i - 1) * length + rev_lengths_sum[i]
                         for i, length in enumerate(rev_lengths)])
```